### ISSUE TITLE
FR: media retention based on access time.

### ISSUE THREAD

**Issue by Kimiblock** [girlbossceo/conduwuit#440]
Purge local and remote media that hasn’t been accessed in a certain amount of time, in order to control the size of media repo. (Preferably automatically)

Synapse manual: https://element-hq.github.io/synapse/latest/usage/configuration/config_documentation.html?highlight=Media_re#media_retention

**Comment by Kimiblock** [comment]
Another approach is to use systemd-tmpfiles with conduwuit's startup check disabled:

```
d	/var/lib/private/serverOS-conduwuit/media	-	-	-	10d
```

This works mostly fine for me, just need some more time to prove the stability. Maybe it can be added to the conduwuit docs.

**Comment by girlbossceo** [comment]
This does not differentiate between local and remote media. This will cause local media to be deleted too.

We plan on separating out the local and remote media into at least two separate directories eventually.

**Comment by Kimiblock** [comment]
Yes, and that's my intention. Users on my server upload media pretty often and the majority of them us useless and never accessed again.

**Comment by tcpipuk** [comment]
Absolutely, and _some_ people will want to purge local media the same way as remote media, but _some_ people will also not want to do that.

Splitting the two and having them configurable separately makes a lot of sense here - it'd let both sets of people do whatever they want! 😊
